File: app/src/main/java/za/org/grassroot2/presenter/fragment/SingleTextMultiButtonPresenter.java
package za.org.grassroot2.presenter.fragment;

import org.greenrobot.eventbus.EventBus;

import io.reactivex.Observable;
import za.org.grassroot2.view.FragmentView;
import za.org.grassroot2.view.event.MoveNextWithInputEvent;


public class SingleTextMultiButtonPresenter extends BaseFragmentPresenter<SingleTextMultiButtonPresenter.SingleTextMultiButtonView> {

    @Override
    public void onViewCreated() {
        disposableOnDetach(getView().inputTextDone().subscribe(charSequence -> {
            EventBus.getDefault().post(new MoveNextWithInputEvent(charSequence.toString()));
        }, Throwable::printStackTrace));
    }

    public interface SingleTextMultiButtonView extends FragmentView {
        Observable<CharSequence> inputTextDone();
        void setupButtons();
    }
}

File: app/src/main/java/za/org/grassroot2/presenter/fragment/SingleTextMultiButtonPresenter.kt
package za.org.grassroot2.presenter.fragment

import org.greenrobot.eventbus.EventBus

import io.reactivex.Observable
import timber.log.Timber
import za.org.grassroot2.view.FragmentView
import za.org.grassroot2.view.event.MoveNextWithInputEvent


class SingleTextMultiButtonPresenter : BaseFragmentPresenter<SingleTextMultiButtonPresenter.SingleTextMultiButtonView>() {

    override fun onViewCreated() {
        disposableOnDetach(view.inputTextDone().subscribe(
                { charSequence -> EventBus.getDefault().post(MoveNextWithInputEvent(charSequence.toString())) }, { Timber.e(it) }))
    }

    interface SingleTextMultiButtonView : FragmentView {
        fun inputTextDone(): Observable<CharSequence>
        fun setupButtons()
    }
}
